### PR TITLE
Garden List initial build

### PR DESCRIPTION
**Pull request by MulesKC**
initial build of models: Feature, Garden, GardenDetails, GardenOwner

    added initial GardenList templates: features/add&list, owner/add&list, and garden/add&list

    updated page-header in fragments.html(fixed logo, added register), updated index.html to use page-header fragment, removed @Entity from models to get program to run (classes weren't complete yet)

    combined Jamie and my initial work on Garden List including: adding Feature,Garden,Owner repositories, removed GardenDetails, updated references to GardenDetails

    added to SecurityConfiguration to post to MySQL, further adjusting templates, controllers, and models to get entry of new features, owners, and gardens working - garden creation not quite finished

    database working correctly; Garden, owner, and feature listing working

    updated gardens/add for latitude & longitude; added validation to some fields in Garden.java

    added '.' for img-fluid in css; removed my quick-fix in SecurityConfig; updated page-header fragment with garden, features, and owners links; added link to 'add garden' on garden list page

    fixed linking to Garden list; added back some code to SecurityConfig (**line 56**) to get database working again; made latitude/longitude strings for now

### CODE REVIEW

**Review by scottpenneman — APPROVED** [review]
Nice work!

File: src/main/resources/templates/gardens/add.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    <div class="form-group">
+        <label th:for="owner">Garden Owner</label>
+        <select name="owner">
+            <option th:each="owner : ${owners}"
+                    th:text="${owner.name}"
+                    th:value="${owner.id}"></option>
+        </select>
+        <a th:href="@{'/owner/add'}">Add Owner</a>
+    </div>
+    <div class="form-group">
```
Comment (hstrodtman) [inline src/main/resources/templates/gardens/add.html:47]: This div does not have a closing tag.

File: src/main/resources/templates/gardens/list.html [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                    </li>
+                </ul>
+                </p>
+                <p>
+                    Owner:
+                    <a th:text="${garden.owner.name}"></a>
+<!--                    <a th:href="@{'/owners/view/' + ${garden.gardenOwner.id}}" th:text="${garden.gardenOwner.name}"></a>-->
+                </p>
+                </hr>
+            </td>
```
Comment (hstrodtman) [inline src/main/resources/templates/gardens/list.html:44]: There is no opening td tag and an extra closing div tag.

**Review by hstrodtman — APPROVED** [review]
Awesome work getting all this done! Just added a few small syntax comments. Also, adding the th:action tag to the form element does seem to solve the security issues.

**Review by jamiander — APPROVED** [review]
Let me know if you have merge conflicts with any of the registration stuff.  I can help so none of the functionality gets funky.  Great job!  Lots of work!

**Pull request merged**